### PR TITLE
testing: add tests

### PR DESCRIPTION
**Pull request by broady**

### CODE REVIEW

**Comment by broady** [comment]
I decided to keep the amount of travis stuff to a minimum, like not using their method to install Go, to make it easy to run these tests on other machines or inside a more minimal Docker container.

**Comment by bshaffer** [comment]
This is awesome! Is `.bash` more popular in the go community? Whichever is used, we should be consistent w/ the `build.sh` file.  Otherwise LGTM

**Comment by broady** [comment]
woohoo! they run on Travis.

File: testing/run_docker.bash [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+set -e -x
+
+SRC=/gopath/src/github.com/GoogleCloudPlatform/appengine-sidecars-docker
+
+docker run --rm -i \
+  -e GOPATH=/gopath \
+  -w $SRC \
+  -v $PWD:$SRC \
+  -t quay.io/travisci/travis-ruby:latest \
```
Comment (dlorenc) [inline testing/run_docker.bash]: Could you use the golang image instead of this, and then avoid the deps install?

**Comment by dlorenc** [comment]
One question, otherwise LGTM.

File: testing/run_docker.bash (hunk 1 above)
Comment (broady) [inline testing/run_docker.bash]: Yes, but I put the deps install there in the hope that we might one day use a cleaner image to run the tests.

Do you think that's important? Probably not.

**Comment by broady** [comment]
Gonna submit in the interest of moving it along. We can revisit later.

**Pull request merged**